dotnet/src/SemanticKernel.Core/Functions/ContextualSelection/FunctionStore.cs

```diff
@@ -50,7 +55,7 @@ public FunctionStore(
         this._collectionName = collectionName;
         this._functionByName = functions.ToDictionary(function => function.Name);
         this._maxNumberOfFunctions = maxNumberOfFunctions;
-
+        this._logger = loggerFactory?.CreateLogger<FunctionStore>() ?? NullLogger.Instance as ILogger;
```

Comment:
nit. Thought `NullLogger.Instance` was already an `ILogger` curious why the cast is needed.

```suggestion
        this._logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<FunctionStore>();
```